### DISCUSSION TITLE
[BUG]: Receipt Object - Status Enum

### DISCUSSION THREAD

**Discussion by gordonturner** [etsy/open-api#732] [Ideas]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

The List of Shop Receipt resources returned in the getShopReceipts has an inconsistency in the definition of the `status` object.

- The definition of the status is:

status
required
string
Enum: "paid" "completed" "open" "payment processing" "canceled"
The current order status string. One of: Open, Paid, Completed, Payment Processing.

- Reference:

https://developer.etsy.com/documentation/reference/#operation/getShopReceipts

The generated code (via the Open API generator) creates a Java enum of the lower case text.

The returned values (and the values in the documentation) are returned with leading capital letters.

Result in an exception b/c of mismatch in the text expected and value returned.

### Steps to reproduce

Generated java code by running:

```
java -jar openapi-generator-cli.jar generate \
-i https://www.etsy.com/openapi/generated/oas/3.0.0.json \
-g java \
-o ./etsy-open-api-client \
--skip-validate-spec \
--group-id com.gordonturner.etsy \
--artifact-id etsy-open-api-client \
--artifact-version 1.1.0 \
--type-mappings=integer=Long,int=Long
```

Observe the enum created at:

org.openapitools.client.model.ShopReceipt.StatusEnum

```
    PAID("paid"),
    
    COMPLETED("completed"),
    
    OPEN("open"),
    
    PAYMENT_PROCESSING("payment processing"),
    
    CANCELED("canceled");
```

### Expected behavior

Expected that the returned value from the API should be lower case (I think).

**Comment by gordonturner** [comment]
Proposed label: OpenAPI

**Comment by etsyachristensen** [comment]
Converting to an Idea discussion because we are closing the issues board down.

**Comment by colesnic89** [comment]
@etsyachristensen  
[there](https://developer.etsy.com/documentation/reference/#operation/getShopReceipts) is no filter by status. No filter - no bug :)

**Reply by Gareth064** [reply]
@etsyachristensen no longer works for Etsy.

What is it you were hoping to point out? 
Yeah there is no specific filter by status but you can filter by paid, shipped & cancelled. That pretty much covers everything? 

Maybe refunded could be a filter too.

**Reply by colesnic89** [reply]
@Gareth064 Thank you very much! ETSY support couldn't explain it for about a week!!!
Solved!

**Reply by Gareth064** [reply]
No worries @colesnic89. That's what the community is here for. We are all still learning though. This API is not easy!
